vectordatasource/transform.py

```diff
@@ -4029,26 +4029,203 @@ def _guess_type_from_network(network):
+    based on a bunch of logic about what kinds of Australian roads exist.
+
+    Returns new (network, ref) values.
     """
 
-    if country_code == 'GB':
-        letter = ref[0]
-        if letter in ('M', 'A', 'B'):
-            return 'GB:%s-road' % (letter,)
+    # grab the prefix, if any, from the ref. we can use this to "back-fill" the
+    # network.
+    prefix, ref = _splitref(ref)
+
+    if network and network.startswith('AU:') and \
+       network[3:] in _AU_NETWORK_IMPORTANCE:
+        # network is already in the form we want!
+        pass
 
-    elif country_code == 'AR':
-        if ref.startswith('RN'):
-            return 'AR:national'
-        elif ref.startswith('RP'):
-            return 'AR:provincial'
+    elif network in _AU_NETWORK_EXPANSION:
+        network = _AU_NETWORK_EXPANSION[network]
 
-    # in the absence of any other more specific information, might as well
-    # just use the country code as the network (similar to what we do above
-    # when back-filling from the operator).
-    return country_code
+    elif prefix in _AU_NETWORK_EXPANSION:
+        # backfill network from ref, if possible. (note that ref must
+        # be non-None, since mz_networks entries have either network or
+        # ref, or both).
+        network = _AU_NETWORK_EXPANSION[prefix]
+
+    return network, ref
+
+
+# CountryNetworkLogic centralises the logic around country-specific road
+# network processing. this allows us to do different things, such as
+# back-filling missing network tag values or sorting networks differently
+# based on which country they are in. (e.g: in the UK, an "M" road is more
+# important than an "A" road, even though they'd sort the other way
+# alphabetically).
+#
+# the different logic sections are:
+#
+# * backfill: this is called as fn(ref) to unpack the ref tag into a list of
+#             (network, ref) tuples to use instead. For example, it's common
+#             to give ref=A1;B2;C3 to indicate multiple networks & shields.
+#
+# * fix: this is called as fn(network, ref) and should fix whatever problems it
+#        can and return the replacement (network, ref). remember! either
+#        network or ref can be None!
+#
+# * sort: this is called as fn(network, ref) and should return a numeric value
+#         where lower numeric values mean _more_ important networks.
+#
+CountryNetworkLogic = namedtuple(
+    'CountryNetworkLogic', 'backfill fix sort')
+CountryNetworkLogic.__new__.__defaults__ = (None,) * len(
+    CountryNetworkLogic._fields)
+
+_COUNTRY_SPECIFIC_ROAD_NETWORK_LOGIC = {
+    'AR': CountryNetworkLogic(
+        backfill=_guess_network_ar,
+    ),
+    'AU': CountryNetworkLogic(
+        backfill=_guess_network_au,
+        fix=_normalize_au_netref,
+        sort=_sort_network_au,
+    ),
+    'GB': CountryNetworkLogic(
+        backfill=_guess_network_gb,
+    ),
+    'US': CountryNetworkLogic(
+        backfill=_do_not_backfill,
+        sort=_sort_network_us,
+    ),
+}
```

Comment:
Great job with the comments and organizing it this way! 👍

Reply:
Thanks!

I was wondering whether you thought it might be better organised as a set of classes? For example:

```python
class GB(CountryRoadNetwork):
  def backfill(self, ref):
    ...
```

Or do you think the extra syntax and state-handling of classes (i.e: `self`) get in the way of understanding what's going on?

Reply:
I prefer the way you have it now. I think it makes it easier to see what's going on, especially because you picked good names for the underlying functions. It'll make it easier to share primitives between them if we ever need to, and because it's more declarative we have more flexibility around how it's organized.